src/include/snapper/dialogs.rb

```diff
@@ -658,19 +446,16 @@ def ShowDialog
       date = timestring(snapshot["date"])
       type = Ops.get_symbol(snapshot, "type", :NONE)
       combo_items = []
-      Builtins.foreach(Snapper.snapshots) do |s|
-        id = Ops.get_integer(s, "num", 0)
+      Snapper.snapshots.map do |s|
```

Comment:
map is here wrong method. in this case I really recommend to use `each_with_object` like
```
combo_items = Snapper.snapshots.each_with_items([]) do |s, combo_items|
```